flink-runtime/src/test/java/org/apache/flink/runtime/security/contexts/LinkageErrorSecurityContextFactory.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.security.contexts;
+
+import org.apache.flink.runtime.security.SecurityConfiguration;
+import org.apache.flink.runtime.security.SecurityContextInitializeException;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Test security context factory class to test out linkage error circumstances.
+ */
+public class LinkageErrorSecurityContextFactory implements SecurityContextFactory {
```

Comment:
The test doesn't work because this is not added to the correct `META-INF.services` file. The line where the exception is thrown is never run.

Reply:
oops. forgot to put it in the right branch. thanks for noticing.